### PR TITLE
Mnaveau/dynamic graph

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

Add the entities around follwoing classes:
- ImpedeanceControl
- CentroidalPDController
- CentroidalForceQPController

## How I Tested

[//]: # "Explain how you tested your changes"

I have loaded the entities in an ipython terminal and initialized them

Todo: add unit-tests for all three classes for:
- C++
- Python bindings
- Entities

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Review by jviereck — APPROVED** [review]
Minor things, overall this looks very good!

File: include/mim_control/dynamic_graph/signal_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      make_signal_string(true, CLASS_NAME, name, signal_type,                  \
+                         crop_underscore(signal_var_name)))
+
+#define define_output_signal(                                                  \
+    signal_var_name, signal_type, signal_dep, callback)                        \
+  signal_var_name(                                                             \
+      boost::bind(callback, this, _1, _2), signal_dep,                         \
+      make_signal_string(false, CLASS_NAME, name, signal_type,                 \
+                         crop_underscore(signal_var_name)))
+// clang-format on
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply include/mim_control/dynamic_graph/signal_utils.hpp:42]: I am not sure what to do to use these elsewhere, maybe a PR in dynamic-graph?

File: src/dynamic_graph/centroidal_pd_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      define_input_signal(kp_com_sin_, "Vector3d"),
+      define_input_signal(kd_com_sin_, "Vector3d"),
+      define_input_signal(kp_base_sin_, "Vector3d"),
+      define_input_signal(kd_base_sin_, "Vector3d"),
+      define_input_signal(actual_com_position_sin_, "Vector3d"),
+      define_input_signal(desired_com_position_sin_, "Vector3d"),
+      define_input_signal(actual_com_velocity_sin_, "Vector3d"),
+      define_input_signal(desired_com_velocity_sin_, "Vector3d"),
+      define_input_signal(actual_base_orientation_sin_, "Vector7d_quat"),
+      define_input_signal(desired_base_orientation_sin_, "Vector7d_quat"),
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply src/dynamic_graph/centroidal_pd_controller.cpp:38]: You are right it should be a quaternion hence 4d

File: src/dynamic_graph/centroidal_force_qp_controller.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                                  << cnt_array_sin_ << forces_sout_);
+}
+
+void CentroidalForceQPController::initialize(int number_endeffectors,
+                                             double friction_coeff,
+                                             double qp_penalty_lin,
+                                             double qp_penalty_ang)
+{
+    force_ctrl_.initialize(
+        number_endeffectors, friction_coeff, qp_penalty_lin, qp_penalty_ang);
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply src/dynamic_graph/centroidal_force_qp_controller.cpp:48]: Yes we should fix it in another PR

**Pull request merged**